File: godbrain_core/factory_core.py
import asyncio
import logging
import json
from openai import OpenAI

logging.basicConfig(level=logging.INFO, format="%(asctime)s - [FACTORY_CORE] - %(message)s")
logger = logging.getLogger("FactoryCore")

LOCAL_LLM_URL = "http://127.0.0.1:8080/v1"
LOCAL_LLM_KEY = "sk-no-key-required"

class Architect:
    def __init__(self):
        self.client = OpenAI(base_url=LOCAL_LLM_URL, api_key=LOCAL_LLM_KEY)
        logger.info("[+] The Architect is online. Awaiting directives.")

    def spawn_agent(self, agent_type, task_description, context):
        """Simulates spawning a specialized sub-agent with a strict system prompt."""
        logger.info(f"[*] Spawning {agent_type} for task: {task_description}")
        
        system_prompts = {
            "Surgeon": "You are The Surgeon. You output ONLY valid PowerShell scripts. You MUST include a `# REASONING:` comment explaining why this script is safe and necessary.",
            "Watcher": "You are The Watcher. You analyze security data. Output ONLY a JSON array of threat vectors.",
                        "Interceptor": "You are The Interceptor. You analyze network traffic and output Windows Firewall rules.",
            "Oracle": "You are The Oracle. You analyze live market data, prediction markets, and physical world APIs. Output ONLY high-probability arbitrage executions or mathematical certainties with >95% win rates."
        }
        
        prompt = system_prompts.get(agent_type, "You are a generic GodBrain helper.")
        
        try:
            # Send task to the local LLM using the specialized persona
            response = self.client.chat.completions.create(
                model="GodBrain-Colibri",
                messages=[
                    {"role": "system", "content": prompt},
                    {"role": "user", "content": f"Context: {context}\nTask: {task_description}"}
                ],
                temperature=0.1
            )
            result = response.choices[0].message.content
            logger.info(f"[+] {agent_type} Task Complete. Output:\n{result}\n")
            return result
        except Exception as e:
            logger.error(f"[-] {agent_type} Failed: {e}")
            return None

    def execute_directive(self, directive):
        """The Architect breaks down a high-level directive into sub-tasks."""
        logger.warning(f"[!] New Directive Received: {directive}")
        
        # In a full implementation, the LLM would dynamically generate this task list.
        # For now, we simulate the breakdown.
        if "disable telemetry" in directive.lower():
            logger.info("[*] Architect breaking down directive into sub-tasks...")
            
            # Step 1: Analyze
            threats = self.spawn_agent(
                "Watcher", 
                "Identify the main executable for Connected User Experiences and Telemetry in Windows 11.",
                "Target: DiagTrack"
            )
            
            # Step 2: Act
            if threats:
                self.spawn_agent(
                    "Surgeon",
                    "Write a script to force-stop and disable the DiagTrack service.",
                    f"Threat Intel: {threats}"
                )

if __name__ == "__main__":
    architect = Architect()
    architect.execute_directive("Disable telemetry on this host.")

File: godbrain_core/cpp_tools/factory_core.cpp
#define WIN32_LEAN_AND_MEAN
#include <windows.h>
#include <iostream>
#include <string>
#include <vector>
#include <map>

// In the Python version, this used the OpenAI python client against the local server.
// In C++, we can directly hook into our run_colibri logic via pipes, or send a basic 
// HTTP POST to http://127.0.0.1:8080/v1/chat/completions using httplib if we want to 
// maintain the OpenAI API format compatibility. Since we are moving to native C++ everywhere,
// we'll simulate the execution flow.

std::string spawn_agent(const std::string& agent_type, const std::string& task_description, const std::string& context) {
    std::cout << "[*] Spawning " << agent_type << " for task: " << task_description << std::endl;
    
    std::map<std::string, std::string> system_prompts = {
        {"Surgeon", "You are The Surgeon. You output ONLY valid PowerShell scripts. You MUST include a `# REASONING:` comment explaining why this script is safe and necessary."},
        {"Watcher", "You are The Watcher. You analyze security data. Output ONLY a JSON array of threat vectors."},
        {"Interceptor", "You are The Interceptor. You analyze network traffic and output Windows Firewall rules."},
        {"Oracle", "You are The Oracle. You analyze live market data, prediction markets, and physical world APIs. Output ONLY high-probability arbitrage executions or mathematical certainties with >95% win rates."}
    };

    std::string prompt = "You are a generic GodBrain helper.";
    if (system_prompts.find(agent_type) != system_prompts.end()) {
        prompt = system_prompts[agent_type];
    }

    std::string full_prompt = prompt + "\n\nContext: " + context + "\nTask: " + task_description + "\nAnswer:";
    
    std::cout << "[*] Instructing native C++ Colibri Hook..." << std::endl;
    // We mock the execution here for the C++ port, as the full `run_colibri` is in main.cpp.
    // In a fully integrated build, factory_core would link against kernel.h/main.cpp.
    
    std::string mock_result;
    if (agent_type == "Watcher") {
        mock_result = "[\"DiagTrack\", \"CompatTelRunner.exe\"]";
    } else if (agent_type == "Surgeon") {
        mock_result = "# REASONING: Stopping DiagTrack blocks Microsoft telemetry.\nStop-Service -Name DiagTrack -Force\nSet-Service -Name DiagTrack -StartupType Disabled";
    } else {
        mock_result = "Execution Complete.";
    }

    std::cout << "[+] " << agent_type << " Task Complete. Output:\n" << mock_result << "\n\n";
    return mock_result;
}

void execute_directive(const std::string& directive) {
    std::cout << "[!] New Directive Received: " << directive << std::endl;
    
    if (directive.find("disable telemetry") != std::string::npos) {
        std::cout << "[*] Architect breaking down directive into sub-tasks..." << std::endl;
        
        std::string threats = spawn_agent(
            "Watcher", 
            "Identify the main executable for Connected User Experiences and Telemetry in Windows 11.",
            "Target: DiagTrack"
        );
        
        if (!threats.empty()) {
            spawn_agent(
                "Surgeon",
                "Write a script to force-stop and disable the DiagTrack service.",
                "Threat Intel: " + threats
            );
        }
    }
}

int main(int argc, char* argv[]) {
    std::cout << "[+] The Architect is online (C++ Native). Awaiting directives." << std::endl;
    
    if (argc < 2) {
        execute_directive("disable telemetry on this host.");
    } else {
        execute_directive(argv[1]);
    }

    return 0;
}